Microsoft.Toolkit.Uwp.UI.Controls.Core/ImageEx/ImageExBase.Source.cs

```diff
@@ -88,15 +97,14 @@ private async void SetSource(object source)
                 return;
             }
 
-            this._tokenSource?.Cancel();
+            _tokenSource?.Cancel();
 
-            this._tokenSource = new CancellationTokenSource();
+            _tokenSource = new CancellationTokenSource();
 
             AttachSource(null);
```

Comment:
Question: is toggling this visual state unconditionally by design? Because if the input is not `null`, doing this here no matter what means we're basically always cycling between loaded -> unloaded -> loading -> loaded whenever a new image is displayed.

I'm thinking maybe we could skip a couple of visual states here and:
1) Only switch to the unloaded state when we're actually setting a `null` source
2) Only set the loading state if we're loading an image asynchronously (that is, in the first two paths of our `LoadImageAsync` method.

It's a minor change but in theory it might help with performance a tiny bit in cases where we have a lot of controls being displayed eg. in a virtualized list, as those would be toggling the target image all the time?
Just an idea 😄